src/App.tsx

```diff
@@ -1,24 +1,124 @@
-import React from 'react'
-import { Redirect, Route, Switch } from 'react-router'
-import { BrowserRouter } from 'react-router-dom'
-import { Page } from './components/base/base'
-import { TopBar } from './components/TopBar'
-import { GlobalStyle } from './global/GlobalStyle'
-import { NotificationsList } from './components/Transactions/History'
-import { DemoTDec } from './pages/DemoTDecConditions'
-
-export function App() {
+import type {
+  MessageKit,
+  PolicyMessageKit,
+  DeployedStrategy,
+  ConditionSet,
+} from "@nucypher/nucypher-ts";
+import React, { useState } from "react";
+import { useEthers } from "@usedapp/core";
+import { ethers } from "ethers";
+
+import { ConditionBuilder } from "./ConditionBuilder";
+import { Encrypt } from "./Encrypt";
+import { Decrypt } from "./Decrypt";
+import { Spinner } from "./Spinner";
+import { StrategyBuilder } from "./StrategyBuilder";
+
+export default function App() {
+  const { activateBrowserWallet, deactivate, account } = useEthers();
+
+  const [loading, setLoading] = useState(false);
+  const [deployedStrategy, setDeployedStrategy] = useState<DeployedStrategy>();
+  const [conditions, setConditions] = useState<ConditionSet>();
+  const [encryptedMessage, setEncryptedMessage] = useState<MessageKit>();
+  const [decryptedMessage, setDecryptedMessage] = useState("");
+  const [decryptionErrors, setDecryptionErrors] = useState<string[]>([]);
+
+  const encryptMessage = (plaintext: string) => {
+    setLoading(true);
+    deployedStrategy!.encrypter.conditions = conditions;
```

Comment:
I must admit that assigning conditions to a `deployedStrategy` like this look weird, not sure what to think about it yet.